From 1b662711f861b3bf3d080577144ad7e2603eccab Mon Sep 17 00:00:00 2001
From: James Kosin <jkosin@apache.org>
Date: Fri, 11 Nov 2011 23:41:08 +0000
Subject: [PATCH] OPENNLP-361: usage of setlocal and endlocal to keep additions
 and changes to environment local, keep OPENNLP_HOME to a short name if set
 outside batch file.  -- Thanks to Aliaksandr Autayeu

git-svn-id: https://svn.apache.org/repos/asf/incubator/opennlp/trunk@1201107 13f79535-47bb-0310-9956-ffa450edef68
---
 opennlp-distr/src/main/bin/opennlp.bat | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/opennlp-distr/src/main/bin/opennlp.bat b/opennlp-distr/src/main/bin/opennlp.bat
index 3c42051a1..4979e3498 100644
--- a/opennlp-distr/src/main/bin/opennlp.bat
+++ b/opennlp-distr/src/main/bin/opennlp.bat
@@ -20,16 +20,25 @@ REM #   under the License.
 REM # Note:  Do not output anything in this script file, any output
 REM #        may be inadvertantly placed in any output files if
 REM #        output redirection is used.
+SETLOCAL
 
 IF "%JAVA_CMD%" == "" (
 	IF "%JAVA_HOME%" == "" (
 		SET JAVA_CMD=java 
 	) ELSE (
+		REM # Keep JAVA_HOME to short-name without spaces
 		FOR %%A IN ("%JAVA_HOME%") DO SET JAVA_CMD=%%~sfA\bin\java
 	)
 )
 
 REM #  Should work with Windows XP and greater.  If not, specify the path to where it is installed.
-IF "%OPENNLP_HOME%" == "" SET OPENNLP_HOME=%~sp0..
+IF "%OPENNLP_HOME%" == "" (
+	SET OPENNLP_HOME=%~sp0..
+) ELSE (
+	REM # Keep OPENNLP_HOME to short-name without spaces
+	FOR %%A IN ("%OPENNLP_HOME%") DO SET OPENNLP_HOME=%%~sfA
+)
 
 %JAVA_CMD% -Xmx4096m -jar %OPENNLP_HOME%\lib\opennlp-tools-*.jar %*
+
+ENDLOCAL
\ No newline at end of file
